### ISSUE TITLE
Numeric module segfaulting when running with pyodbc==3.08 

### ISSUE THREAD

**Issue by StephanErb** [mkleehammer/pyodbc#37]
We are having issues with the latest pyodbc==3.08 on Ubuntu 12.04 and Python 2.7.

Decimal is working as expected:
```python
Python 2.7.3 (default, Dec 18 2014, 19:10:20) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import decimal
>>> 
KeyboardInterrupt
>>> 
```

However, when running in a virtualenv with the latest pyodbc, decimal can no longer be imported:

```python
Python 2.7.3 (default, Dec 18 2014, 19:10:20) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyodbc
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Unable to import cdecimal or decimal
>>> import decimal
Segmentation fault
```

The relevant pyodbc compilation/installation:
```python
pip install pyodbc==3.0.8
Downloading/unpacking pyodbc==3.0.8
  Downloading pyodbc-3.0.8.tar.gz (70kB): 70kB downloaded
  Running setup.py (path:/tmp/pyodbc-old/build/pyodbc/setup.py) egg_info for package pyodbc
    
    warning: no files found matching 'tests/*'
Installing collected packages: pyodbc
  Found existing installation: pyodbc 3.0.7
    Uninstalling pyodbc:
      Successfully uninstalled pyodbc
  Running setup.py install for pyodbc
    building 'pyodbc' extension
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/pyodbcmodule.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/pyodbcmodule.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/errors.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/errors.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/row.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/row.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/sqlwchar.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/sqlwchar.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/buffer.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/buffer.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/connection.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/connection.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/pyodbccompat.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/pyodbccompat.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/getdata.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/getdata.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    /tmp/pyodbc-old/build/pyodbc/src/getdata.cpp: In function ‘PyObject* GetDataDecimal(Cursor*, Py_ssize_t)’:
    /tmp/pyodbc-old/build/pyodbc/src/getdata.cpp:487:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/pyodbcdbg.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/pyodbcdbg.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/params.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/params.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/cursor.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/cursor.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=3.0.8 -DPYODBC_UNICODE_WIDTH=4 -DSQL_WCHART_CONVERT=1 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include -I/usr/include/python2.7 -c /tmp/pyodbc-old/build/pyodbc/src/cnxninfo.cpp -o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/cnxninfo.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]
    g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/pyodbcmodule.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/errors.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/row.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/sqlwchar.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/buffer.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/connection.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/pyodbccompat.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/getdata.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/pyodbcdbg.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/params.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/cursor.o build/temp.linux-x86_64-2.7/tmp/pyodbc-old/build/pyodbc/src/cnxninfo.o -lodbc -o build/lib.linux-x86_64-2.7/pyodbc.so
    
    warning: no files found matching 'tests/*'
Successfully installed pyodbc
Cleaning up...
```

**Comment by keitherskine** [comment]
I was finding something similar when running the sqlservertests.py script in directory tests3, although I was getting this even with pyodbc 3.0.7.  To get that test script to run, I had to move the line ```from decimal import Decimal``` (line 30) to just after the line that imports pyodbc itself (line 1303).

In my case, I was building from a Windows 7 64-bit PC, using Visual C++ 2010 Express.

**Comment by mkleehammer** [comment]
I believe this is fixed in 3.0.9.  Can you confirm?
